From owner-linux-pci@atrey.karlin.mff.cuni.cz Thu Jul 13 11:00:49 2006
Message-Id: <20060713165609.657546000@myri.com>
Date: Thu, 13 Jul 2006 12:56:17 -0400
From: Brice Goglin <brice@myri.com>
To: linux-pci@atrey.karlin.mff.cuni.cz
Cc: Brice Goglin <brice@myri.com>
Subject: MSI: Drop pci bus_flags
Content-Disposition: inline; filename=msi-09-drop_pci_bus_flags.patch

PCI_BUS_FLAGS_NO_MSI is not used anymore, and gcc does not like
empty enums. Drop the bus_flags field entirely.

Signed-off-by: Brice Goglin <brice@myri.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/probe.c |    1 -
 include/linux/pci.h |    6 ------
 2 files changed, 7 deletions(-)

--- gregkh-2.6.orig/drivers/pci/probe.c
+++ gregkh-2.6/drivers/pci/probe.c
@@ -351,7 +351,6 @@ pci_alloc_child_bus(struct pci_bus *pare
 	child->parent = parent;
 	child->ops = parent->ops;
 	child->sysdata = parent->sysdata;
-	child->bus_flags = parent->bus_flags;
 	child->bridge = get_device(&bridge->dev);
 
 	child->class_dev.class = &pcibus_class;
--- gregkh-2.6.orig/include/linux/pci.h
+++ gregkh-2.6/include/linux/pci.h
@@ -94,11 +94,6 @@ enum pci_channel_state {
 	pci_channel_io_perm_failure = (__force pci_channel_state_t) 3,
 };
 
-typedef unsigned short __bitwise pci_bus_flags_t;
-enum pci_bus_flags {
-	PCI_BUS_FLAGS_NO_MSI = (__force pci_bus_flags_t) 1,
-};
-
 struct pci_cap_saved_state {
 	struct hlist_node next;
 	char cap_nr;
@@ -242,7 +237,6 @@ struct pci_bus {
 	char		name[48];
 
 	unsigned short  bridge_ctl;	/* manage NO_ISA/FBB/et al behaviors */
-	pci_bus_flags_t bus_flags;	/* Inherited by child busses */
 	struct device		*bridge;
 	struct class_device	class_dev;
 	struct bin_attribute	*legacy_io; /* legacy I/O for this bus */
